Products/CMFPlone/recyclebin.py

```diff
@@ -166,6 +169,104 @@ def is_enabled(self):
         except (KeyError, AttributeError):
             return False
 
+    def check_permission(self, check_roles=True):
+        """Check if the current user has permission to manage the recycle bin
+
+        Args:
+            check_roles: If True, also check for Manager or Site Administrator roles
+
+        Returns:
+            Boolean indicating permission
+        """
+        context = self._get_context()
+        sm = getSecurityManager()
+
+        # Check for explicit permission first
+        has_permission = sm.checkPermission(self.MANAGE_RECYCLEBIN, context)
+        if has_permission:
+            return True
+
+        # If no explicit permission and we're not checking roles, return False
+        if not check_roles:
+            return False
+
+        # Additional check for Manager or Site Administrator roles
+        user = sm.getUser()
+        user_roles = user.getRolesInContext(context)
+        return "Manager" in user_roles or "Site Administrator" in user_roles
```

Comment:
Remove the explicit check for roles. That's already included in the checkPermission call (it checks that the current user has one of the roles for which the permission has been granted)